### PR TITLE
Default user name isn't necessarily the same as user attr name

### PR DESCRIPTION
**Pull request by KoviRobi**
Me with my weird system again, sorry.

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by nzbr** [comment]
@KoviRobi I have opened a PR against yours that adds tests for this https://github.com/KoviRobi/NixOS-WSL/pull/2

**Comment by SuperSandro2000** [comment]
I am actually the same so I am surprised that it works for me and not for you.

File: modules/wsl-distro.nix [hunk 1]
```diff
@@ -153,7 +156,7 @@ with lib; {
         })
         (mkIf cfg.nativeSystemd {
           wsl.wslConf = {
-            user.default = cfg.defaultUser;
+            user.default = config.users.users.${cfg.defaultUser}.name;
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:159]: ah, probably because my user.users attr matches name.

**Review by SuperSandro2000 — APPROVED** [review]

File: modules/wsl-distro.nix (hunk 1 above)
Comment (nzbr) [reply modules/wsl-distro.nix:159]: Changing the username has always worked fine, but having different username and attr hasn't

**Pull request merged**